components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/CertificatesApiServiceImpl.java

```diff
@@ -0,0 +1,282 @@
+
+        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        int tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+        String userName = RestApiUtil.getLoggedInUsername();
+
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            if (!apiProvider.isCertificatePresent(tenantId, alias)) {
+                String message = "Certificate for alias '" + alias + "' is not found.";
+                RestApiUtil.handleResourceNotFoundError(message, log);
+            }
+
+            int responseCode = apiProvider.deleteCertificate(userName, alias, null);
+
+            if (responseCode == ResponseCode.SUCCESS.getResponseCode()) {
+                return Response.ok().entity("The certificate for alias '" + alias + "' deleted successfully.").build();
+            } else {
+                RestApiUtil.handleInternalServerError("Error while deleting the certificate for alias '" +
+                        alias + "'.", log);
+            }
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while deleting the certificate for alias '" +
+                    alias + "'.", e, log);
+        }
+        return null;
+    }
+
+    @Override
+    public Response certificatesAliasGet(String alias) {
+
+        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        int tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+
+        if (!StringUtils.isNotEmpty(alias)) {
+            RestApiUtil.handleBadRequest("The alias cannot be empty", log);
+        }
+
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            if (!apiProvider.isCertificatePresent(tenantId, alias)) {
+                String message = "Certificate for Alias '" + alias + "' is not found.";
+                RestApiUtil.handleResourceNotFoundError(message, log);
+            }
+
+            CertificateInformationDTO certificateInformationDTO = apiProvider.getCertificateStatus(alias);
+
+            CertificateValidityDTO certificateValidityDTO = new CertificateValidityDTO();
+            certificateValidityDTO.setFrom(certificateInformationDTO.getFrom());
+            certificateValidityDTO.setTo(certificateInformationDTO.getTo());
+
+            CertificateInfoDTO certificateInfoDTO = new CertificateInfoDTO();
+            certificateInfoDTO.setValidity(certificateValidityDTO);
+            certificateInfoDTO.setStatus(certificateInformationDTO.getStatus());
+            certificateInfoDTO.setSubject(certificateInformationDTO.getSubject());
+            certificateInfoDTO.setVersion(certificateInformationDTO.getVersion());
+
+            return Response.ok().entity(certificateInfoDTO).build();
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while retrieving the certificate status.", e, log);
+        }
+        return null;
```

Comment:
returning null may lead to null pointer issues.

Reply:
This return statement will not be executed as all the return possibilities are handled in the implementation logic.